hack/update-go-modules.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script tidies all go modules in the repository.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+
+mapfile -t DIRS < <(find "${REPO_ROOT}" -name go.mod -print0 | xargs -0 dirname)
```

Comment:
`dirname` terminates its output with a newline, right?  So despite all the effort, this still fails when a directory name includes a newline, right?  There are simpler approaches that accomplish as much.  OTOH, if you want to succeed when the directory name contains a newline, perhaps `find ... -exec process.sh \{\} \;`.

Reply:
I am following the shellcheck suggestions

Reply:
The following make shellcheck happy and works with newlines in the directory name.  It starts with a listing showing all the paths used in this test.

```bash-5.1$ find . -name go.mod
./ta
ste/go.mod
./go.mod
./docs/go.mod
./te  st/go.mod
./pkg/apis/go.mod

bash-5.1$ cat > process.sh
#!/usr/bin/env bash
echo "Working on (${1%/go.mod})"

bash-5.1$ chmod a+x process.sh

bash-5.1$ find . -name go.mod -exec "$PWD/process.sh" \{\} \;
Working on (./ta
ste)
Working on (.)
Working on (./docs)
Working on (./te  st)
Working on (./pkg/apis)
```